### PR TITLE
Added wishlist management

### PR DESCRIPTION
**Pull request by JaanviNL**

### CODE REVIEW

File: src/Pages/Product.jsx [hunk 1]
```diff
@@ -1,30 +1,25 @@
 import "../styles/store.css";
 import "../components/Headerhome.css";
-// import ProductImage from "../assests/p-1.webp";
-import { useEffect, useState } from "react";
-import axios from 'axios';
+
 import Filters from "../components/Filters";
 
-export default function Product() {
+import { useProduct } from "../context/product-context";
```
Comment (Rohitprasad83) [inline src/Pages/Product.jsx:6]: use barrel imports.

File: src/context/product-context.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+function ProductProvider({ children }) {
+
+    const [product, setProduct] = useState([]);
+
+  useEffect(() => {
+    const getProducts = async () => {
+        const res = await axios.get("/api/products")
+        
+        setProduct(res.data.products)
+        console.log(res.data.products)
```
Comment (Rohitprasad83) [inline src/context/product-context.js:17]: remove console.logs

File: src/context/product-context.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    const [product, setProduct] = useState([]);
+
+  useEffect(() => {
+    const getProducts = async () => {
+        const res = await axios.get("/api/products")
+        
+        setProduct(res.data.products)
+        console.log(res.data.products)      
+    }
```
Comment (Rohitprasad83) [inline src/context/product-context.js:18]: add error handling if the api request fails

**Review by Rohitprasad83 — APPROVED** [review]
please format the code as there are empty spaces and also the indentation is not accurate in files like index.js.

File: src/Pages/Product.jsx (hunk 1 above)
Comment (JaanviNL) [reply src/Pages/Product.jsx:6]: okay.

File: src/Pages/Product.jsx [hunk 4]
```diff
@@ -1,30 +1,25 @@
...
+ 
   return (
     <div className="product-page">
      <Filters/>
       
       <div className="products">
+        
         <div className="f-row">
           <p> Products (Showing {product.length} Products)</p>
           <select name="" id="" className="drop-down">
```
Comment (surbhi84) [inline src/Pages/Product.jsx:25]: remove name and id attributes if you're not using them

File: src/Pages/Product.jsx [hunk 5]
```diff
@@ -36,11 +31,15 @@ export default function Product() {
         </div>
         
         <div className="products-card">
-        
-    
-      {product && product.map((item) => <div className="product-card">
+        {product && product.map((item) => <div className="product-card">
             <span>
-              <i className="far fa-heart card-icon"></i>
+               <button className="card-wishlist-icon" onClick={()=>onclickaddwishlist(item)}>
```
Comment (surbhi84) [inline src/Pages/Product.jsx:36]: you should use camel casing for function names makes it more readable.

**Review by surbhi84 — COMMENTED** [review]
The logic is straightforward and the file structure looks clean.

**Pull request merged**